lib/manager/batect/extract.ts

```diff
@@ -0,0 +1,72 @@
+import { safeLoad } from 'js-yaml';
+
+import { logger } from '../../logger';
+import { PackageFile } from '../common';
+import { getDep } from '../dockerfile/extract';
+
+interface BatectConfig {
+  containers?: Record<string, BatectContainer>;
+}
+
+interface BatectContainer {
+  image?: string;
+}
+
+function loadConfig(content: string): BatectConfig {
+  const config = safeLoad(content);
+
+  if (typeof config !== 'object') {
+    throw new Error(
+      `Configuration file does not contain a YAML object (it is ${typeof config}).`
+    );
+  }
+
+  return config as BatectConfig;
+}
+
+function uniqueValues<T>(items: T[]): T[] {
+  return [...new Set(items)];
+}
+
+function extractImages(config: BatectConfig): string[] {
+  if (config.containers === undefined) {
+    return [];
+  }
+
+  const images = Object.values(config.containers)
+    .filter((container) => container.image !== undefined)
+    .map((container) => container.image);
+
+  return uniqueValues(images);
```

Comment:
I'm not sure you want this uniqueness filtering. e.g. if there's the same image mentioned twice in one config file then you'd want a deps of length 2 so that Renovate knows to update it in two places and not one. Can you verify this against a real repo?

Reply:
Yea, please do not deduplicate.

```suggestion
  return images;
```